sentry-apollo-3/src/main/java/io/sentry/apollo3/SentryApollo3HttpInterceptor.kt

```diff
@@ -0,0 +1,162 @@
+package io.sentry.apollo3
+
+import com.apollographql.apollo3.api.http.HttpRequest
+import com.apollographql.apollo3.api.http.HttpResponse
+import com.apollographql.apollo3.exception.ApolloNetworkException
+import com.apollographql.apollo3.network.http.HttpInterceptor
+import com.apollographql.apollo3.network.http.HttpInterceptorChain
+import io.sentry.Breadcrumb
+import io.sentry.Hint
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.ISpan
+import io.sentry.SentryLevel
+import io.sentry.SpanStatus
+import io.sentry.TracingOrigins
+import io.sentry.TypeCheckHint
+import okio.Buffer
+import java.io.InputStreamReader
+
+class SentryApollo3HttpInterceptor(private val hub: IHub = HubAdapter.getInstance(), private val beforeSpan: BeforeSpanCallback? = null) :
```

Comment:
We need the ctor with only hub and only `BeforeSpanCallback` as well due to Java compatibility